src/core/manager.c

```diff
@@ -3264,6 +3267,38 @@ void manager_override_watchdog(Manager *m, WatchdogType t, usec_t timeout) {
         m->watchdog_overridden[t] = timeout;
 }
 
+void manager_set_watchdog_pretimeout_governor(Manager *m, const char *governor) {
+
+        assert(m);
+
+        if (MANAGER_IS_USER(m))
+                return;
+
+        if (streq_ptr(m->watchdog_pretimeout_governor, governor))
+                return;
+
+        if (watchdog_setup_pretimeout_governor(governor) < 0)
+                return;
+
+        free_and_strdup(&m->watchdog_pretimeout_governor, governor);
```

Comment:
maybe strdup() the string before issuing watchdog_setup_pretimeout_governor(), and fail then?